lib/pysquared/logger.py

```diff
@@ -3,15 +3,51 @@
 Logs can be output to standard output or saved to a file (functionality to be implemented).
 """
 
+# NVM register number
 import json
 import time
 
+from lib.pysquared.nvm.counter import Counter
+from lib.pysquared.nvm.registers import ERRORCNT
+
+try:
+    from stubs.circuitpython.byte_array import ByteArray
+except ImportError:
+    pass
+
+
+class LogMode:
+    PRINT = 1
+    FILE = 2
+    BOTH = 3
+
+
+class LogLevel:
+    NOTSET = 0
+    DEBUG = 1
+    INFO = 2
+    WARNING = 3
+    ERROR = 4
+    CRITICAL = 5
+
 
 class Logger:
-    def __init__(self) -> None:
+    def __init__(
+        self,
+        datastore: ByteArray,
+        log_level: int = LogLevel.DEBUG,
+        log_mode: int = LogMode.PRINT,
+    ) -> None:
+        self.datastore: ByteArray = datastore
         self.logToStandardOut: bool = True
```

Comment:
A nitpick, but this is the only thing in this file that is camel case rather than snake case. 
```suggestion
        self.log_to_standard_out: bool = True
```